### PR TITLE
[macOS] Fix build on pre 10.12 and deprecation warnings on 10.12+

### LINKED ISSUE #15376: Compilation error on OS X El Capitan
**Godot version:**
Godot 3 ([b1ab44a](https://github.com/godotengine/godot/commit/b1ab44aa05994a19f0bae5e700c4007b870a8ecf))

**OS/device including version:**
OS X 10.11.6

**Issue description:**
Godot doesn't compile on El Capitan due to an undeclared identifier introduced by #15033

```
platform/osx/os_osx.mm:108:32: error: use of undeclared identifier 'NSEventModifierFlagCommand'
                if (([event modifierFlags] & NSEventModifierFlagCommand) && [event keyCode] == 0x2f) {
```

I think `NSEventModifierFlagCommand` is only available in macOS 10.12+.

Is it macOS Sierra the minimum supported version?

### PR DESCRIPTION
**Pull request by bruvzg**
Constants changed to 10.12+ version, added defines for pre 10.12
NSInputManager replaced with NSTextInputContext

Fixes deprecation warnings.
Fixes #15376.

### CODE REVIEW

File: platform/osx/os_osx.mm [hunk 1]
```diff
@@ -390,8 +401,13 @@ - (NSRect)firstRectForCharacterRange:(NSRange)aRange actualRange:(NSRangePointer
 
 - (void)cancelComposition {
 	[self unmarkText];
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
```
Comment (akien-mga) [inline platform/osx/os_osx.mm]: Our minimum support version is 10.9 according to `detect.py`, so I'm not sure this is necessary.

**Comment by bruvzg** [comment]
Since min. version is set to 10.9 I have removed all pre 10.7 code.

**Comment by icnikerazah** [comment]
can i compile latest right now, because a195f81a6 build doesn't work.

**Comment by bruvzg** [comment]
FYI: Installed OSX 10.9.5 in VirtualBox and did some testing: 

---

Latest master built on 10.13.2 (17C88) with Xcode 9.2 (9C40B) can run on 10.9.5 (13F34) but crashes on exit with:
```
dyld: lazy binding failed: Symbol not found __os_log_internal
```
https://github.com/godotengine/godot/blob/master/platform/osx/os_osx.mm#L1194-L1243

---

Latest master can't be built with Xcode 6.2 (6C131E) (last supported Xcode version on 10.9), build fails with:
```
core/reference.h:261:29: error: member access into incomplete type 'EditorHelpSearch::IncrementalSeatch'
     if (reference && reference->unreference()) {
                               ^
```

---

Latest master can be built with macports clang-5.0.1 (with `macports_clang=5.0` SCons flag) and 10.9 SDK from Xcode 6.2 on OSX 10.9.5 and works fine and much faster then 10.13.2 built binary.

**Pull request merged**